apis/config/v1beta1/configuration_types.go

```diff
@@ -199,6 +202,20 @@ type WaitForPodsReady struct {
 	RequeuingTimestamp *RequeuingTimestamp `json:"requeuingTimestamp,omitempty"`
 }
 
+type MultiKueue struct {
+	// GCTimeout defines the timeout between two consecutive garbage collection runs.
+	// Defaults to 1min. If 0, the garbage collection is disabled.
+	// +optional
+	GCTimeout *metav1.Duration `json:"gcTimeout,omitempty"`
```

Comment:
Why a pointer? I see two alternatives:
1. make it value (no ptr), interpret 0 as disabled
2. make it ptr, validate > 0, interpret null as disabled

Both are fine to me. I find (2.) more readable, but this only slight preference, because it means extra validation.

Reply:
nil is interpreted as not specified in the config and subject to defaulting.
0  is disabled.

Reply:
Ok, but if we do defaulting anyway why not just get rid of the *?

Reply:
If it's not a pointer we will not know if the user did not specified the value and wants to go with the default or the user wants the GC disabled.

Reply:
If it's not a pointer we will not know if the user did not specified the value and wants to go with the default or the user wants the GC disabled.

Reply:
Ah, I think I know what you mean, so that on existing on upgrade the user may set:
```
multikueue:
  origin: multikeue-a
```
forgetting to set the interval param, then it will be left as 0, which would be generally not desired. This makes sense, but it is tricky. My option to has similar issue - we would get it disabled, which is probably not the intention. The intention is most likely to use the default.

I think the current version make sense, otherwise we would need something like `gcEnabled` extra field, but might be overkill.